modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -518,6 +505,25 @@ public CompletableFuture<Void> closeAsync() {
         return stage;
     }
 
+    private static QueryTransactionWrapper beginImplicitTxIfNeeded(
```

Comment:
I think such method should be part of QueryTransactionWrapper. And the QueryTransactionWrapper shouldn't have public constructors

Reply:
:weary: But as far as I remember, we decided that this is wrong from an architectural point of view.

We decided that wrapper should not do any unclear things :roll_eyes: .

For example what is the return type of the following call?

wrapper.beginImplicitTxIfNeeded(..

`void` or `InternalTransaction` or... 'QueryTransactionWrapper`?

Reply:
it should be factory method `QueryTransactionWrapper`

Reply:
Done https://github.com/apache/ignite-3/pull/2449/files#diff-d92eb2bb714b57a42aa2c745c30005875861cc19234e6e29cf5ed0a1b0973bdaR37